src/state/quick-bar-mixin.ts

```diff
@@ -15,9 +15,36 @@ const isMacOS = /(Mac|iPhone|iPod|iPad)/i.test(navigator.platform);
 
 export default <T extends Constructor<HassElement>>(superClass: T) =>
   class extends superClass {
+    private _isAdmin = false;
+
+    private _haveCheckedAdminStatus = false;
+
+    protected updated(changedProps: PropertyValues) {
+      if (this._haveCheckedAdminStatus) {
+        return;
+      }
+
+      if (changedProps.has("hass")) {
+        if (this.hass?.user) {
+          this.isAdmin = this.hass?.user?.is_admin || false;
+          this.haveCheckedAdminStatus = true;
+
+          if (this.isAdmin) {
+            this.registerShortcut();
+          }
+        }
+      }
+    }
+
     protected firstUpdated(changedProps: PropertyValues) {
       super.firstUpdated(changedProps);
 
+      if (this._isAdmin) {
+        this._registerShortcut();
+      }
+    }
+
+    private _registerShortcut() {
       document.addEventListener("keydown", (e: KeyboardEvent) => {
         if (this.isOSCtrlKey(e) && e.code === "KeyP") {
```

Comment:
If the timing is so hard, let's just add a `if (!this.hass?.user?.is_admin) { return; }` here.